File: src/main/java/s4j/java/chapter13/ShoppingBasket.java
package s4j.java.chapter13;

import java.util.HashSet;
import java.util.Set;

public class ShoppingBasket {

    private final Set<Item> basket = new HashSet<>();

    public void add(Item item) {
        basket.add(item);
    }

    public Double value() {
        return basket.stream().mapToDouble(Item::price).sum();
    }
}

File: src/main/scala/s4j/scala/chapter20/ShoppingBasket.scala
package s4j.scala.chapter20

import scala.collection.mutable

class ShoppingBasket {
  private val basket = mutable.HashSet[Item]()

  def add(items: Item*) {
    for (item <- items)
      basket.add(item)
  }

  def value: Double = {
    basket.map(_.price).sum
  }
}
